src/basic/env-util.c

```diff
@@ -779,7 +779,7 @@ int serialize_environment(FILE *f, char **environment) {
                 if (!ce)
                         return -ENOMEM;
 
-                fprintf(f, "env=%s\n", *e);
+                fprintf(f, "env=%s\n", ce);
```

Comment:
Yeah, that's a bummer. But on some level it is justifiable — `--property` uses the same syntax as the unit files, and `set-environment` just gets the final value. The alternative would be to disable escaping in `--property` and also make it take the final value. That would work, because you can always use native shell escaping to craft the argument to `--property`, but I think it'd be surprising that the Environment=... syntax is different in this case than in the unit files. I think that would be worse. We should probably add a note in the docs that things are like that.